axiom/monitors.go

```diff
@@ -11,6 +12,61 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
+//go:generate go run golang.org/x/tools/cmd/stringer -type=Operator -linecomment -output=monitors_string.go
+
+// Operator represents a comparison operation for a monitor. A monitor acts on
+// the result of comparing a query result with a threshold.
+type Operator uint8
+
+// All available monitor comparison modes.
+const (
+	emptyOperator Operator = iota //
+
+	Below        // Below
+	BelowOrEqual // BelowOrEqual
+	Above        // Above
+	AboveOrEqual // AboveOrEqual
+)
+
+func comparisonFromString(s string) (c Operator, err error) {
+	switch s {
+	case emptyOperator.String():
+		c = emptyOperator
+	case Below.String():
+		c = Below
+	case BelowOrEqual.String():
+		c = BelowOrEqual
+	case Above.String():
+		c = Above
+	case AboveOrEqual.String():
+		c = AboveOrEqual
+	default:
+		err = fmt.Errorf("unknown comparison %q", s)
+	}
+
+	return c, err
+}
+
+// MarshalJSON implements `json.Marshaler`. It is in place to marshal the
```

Comment:
This might be a relict from the old codebase but while you're at it and there are conflicts anyways... Could you replace ``json.Marshaler`` with `[json.Marshaler]`. Essentially just making sure that backticks are replaced with matching square brackets.